sycl/test/scheduler/CommandCleanupThreadSafety.cpp

```diff
@@ -0,0 +1,38 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lpthread
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+#include <CL/sycl.hpp>
+
+#include <cassert>
+#include <cstddef>
+#include <thread>
+#include <vector>
+
+// This test checks that the command graph cleanup works properly when
+// invoked from multiple threads.
+using namespace cl::sycl;
+
+class Foo;
```

Comment:
Probably it should be better to move to ./unittests/thread_safety for consistency. But I'm OK with LIT test.

Reply:
This will be addressed later

Reply:
Ok